overlord/hookstate/context.go

```diff
@@ -56,13 +57,45 @@ func NewContext(task *state.Task, setup *HookSetup, handler Handler) (*Context,
 
 	return &Context{
 		task:    task,
+		state:   task.State(),
 		setup:   setup,
 		id:      base64.URLEncoding.EncodeToString(idBytes),
 		handler: handler,
 		cache:   make(map[interface{}]interface{}),
 	}, nil
 }
 
+// NewSnapContext returns a new snap Context.
+func NewSnapContext(state *state.State, setup *HookSetup, handler Handler) (*Context, error) {
+	// Generate a secure, random ID for this context
+	idBytes := make([]byte, 32)
+	_, err := rand.Read(idBytes)
+	if err != nil {
+		return nil, fmt.Errorf("cannot generate context ID: %s", err)
+	}
+
+	return &Context{
+		task:    nil,
```

Comment:
this means things like Context.Get/Set will explode,  I wonder whether we should have a Context interface with the shared bits and two implementations SnapContext and TaskContext, hmm